xcube/core/store/fs/impl/geotiff.py

```diff
@@ -58,27 +59,43 @@ def __init__(self,
         self._root = root
         self._path = data_id
         self._open_params = open_params
+        self._file_url = None
 
-    def _get_num_levels_lazily(self) -> int:
-        with rasterio.open(self._get_file_url()) as rio_dataset:
+    def _open_dataset_with_rasterio(self):
```

Comment:
This name is missleading, rename please
```suggestion
    def _get_overview_count(self):
```